src/Utils.jl

```diff
@@ -147,13 +147,7 @@ array. The use of `@turbo` macro gives a significant performance boost.
 - `best`: The index of the minimum value in the `dij` array.
 """
 fast_findmin(dij, n) = begin
-    # findmin(@inbounds @view dij[1:n])
-    best = 1
-    @inbounds dij_min = dij[1]
-    @turbo for here in 2:n
-        newmin = dij[here] < dij_min
-        best = newmin ? here : best
-        dij_min = newmin ? dij[here] : dij_min
-    end
-    dij_min, best
+    x = @fastmath foldl(min, dij)
```

Comment:
so I think the error is telling us that:
1. there are duplicated value in the distance vector
2. and this new algorithm is picking a different element to return than the current behavior

Interestingly, removing `fastmath` doesn't change the result, and it doesn't help if we change `findfirst` -> `findlast`

I'm a bit lost, what was LoopVectorization doing?